provider/provider_test.go

```diff
@@ -587,3 +675,290 @@ func TestCanPreview(t *testing.T) {
+					"context":    resource.NewStringProperty("."),
+				}),
+			},
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("docker.io/foo/bar:latest"),
+				"build": resource.NewObjectProperty(resource.PropertyMap{
+					"dockerfile": resource.NewStringProperty("Dockerfile"),
+					"context":    resource.NewStringProperty("."),
+				}),
+			},
+			want: &rpc.DiffResponse{
+				Changes: rpc.DiffResponse_DIFF_NONE,
+			},
+		},
+		{
+			name: "Changes in build context",
+			olds: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("docker.io/foo/bar:latest"),
+				"build": resource.NewObjectProperty(resource.PropertyMap{
+					"dockerfile": resource.NewStringProperty("Dockerfile"),
+					"context":    resource.NewStringProperty("."),
+				}),
+			},
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("docker.io/foo/bar:latest"),
+				"build": resource.NewObjectProperty(resource.PropertyMap{
+					"dockerfile": resource.NewStringProperty("Dockerfile"),
+					"context":    resource.NewStringProperty("./new-context"),
+				}),
+			},
+			want: &rpc.DiffResponse{
+				Changes: rpc.DiffResponse_DIFF_SOME,
+				DetailedDiff: map[string]*rpc.PropertyDiff{
+					"build": {
+						Kind: rpc.PropertyDiff_UPDATE,
+					},
+				},
+				HasDetailedDiff: true,
+			},
+		},
+		{
+			name: "Changes in imageName",
+			olds: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("docker.io/foo/bar:latest"),
+			},
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("docker.io/foo/baz:latest"),
+			},
+			want: &rpc.DiffResponse{
+				Changes: rpc.DiffResponse_DIFF_SOME,
+				DetailedDiff: map[string]*rpc.PropertyDiff{
+					"imageName": {
+						Kind: rpc.PropertyDiff_UPDATE,
+					},
+				},
+				HasDetailedDiff: true,
+			},
+		},
+		{
+			name: "Changes in registry server",
+			olds: resource.PropertyMap{
+				"registry": resource.NewObjectProperty(resource.PropertyMap{
+					"server": resource.NewStringProperty("dockerhub"),
+				}),
+			},
+			news: resource.PropertyMap{
+				"registry": resource.NewObjectProperty(resource.PropertyMap{
+					"server": resource.NewStringProperty("ShinyPrivateGHCR"),
+				}),
+			},
+			want: &rpc.DiffResponse{
+				Changes: rpc.DiffResponse_DIFF_SOME,
+				DetailedDiff: map[string]*rpc.PropertyDiff{
+					"registry": {
+						Kind: rpc.PropertyDiff_UPDATE,
+					},
+				},
+				HasDetailedDiff: true,
+			},
+		},
+		{
+			name: "No changes on password update",
+			olds: resource.PropertyMap{
+				"registry": resource.NewObjectProperty(resource.PropertyMap{
+					"password": resource.NewStringProperty("oldpassword"),
+				}),
+			},
+			news: resource.PropertyMap{
+				"registry": resource.NewObjectProperty(resource.PropertyMap{
+					"password": resource.NewStringProperty("newpassword"),
+				}),
+			},
+			want: &rpc.DiffResponse{
+				Changes: rpc.DiffResponse_DIFF_NONE,
```

Comment:
Ah so to clarify you want to completely hide the changes? Displaying the changes without causing a cascading replacement is undesirable? Why? Does not the user expect to see changes reflected after they are made?

Reply:
Those are settings that are likely to change on every run (e.g. for ECR). We’re already doing this for the `Image` resource. This change aligns the bridged resources with that.

without this, drift detection wouldn’t work for most major clouds

Reply:
Gotcha so we want this as default behavior.